### ISSUE TITLE
Version 1.28 stuck on "Validating Server" under MacOS 10.12.5 (Sierra)

### ISSUE THREAD

**Issue by ghoetker** [amaxwell/tlutility#53]
Just installed MacTex, including TeX Live Utility. Upon starting the program, it sticks on "Validating server". I have uninstalled (using AppCleaner to also get preferences, caches, etc.) and then reinstalled the binary from Github. It yields the same issue. 

Please let me know if I can provide any additional information.

**Comment by amaxwell** [comment]
It should time out, eventually. The server validation is probably trying a mirror that's unavailable. How long did you wait for "validating server?"

**Comment by ghoetker** [comment]
Hi. Tried again today. After 2 hours, it was still stuck.

Glenn

Glenn P. Hoetker
Arizona State University | W. P. Carey School of Business
Professor & Dean's Council Distinguished Scholar
Affiliate Professor   Sandra Day O'Conner College of Law
Senior Sustainability Scholar   Julie Ann Wrigley Global Institute of Sustainability
Faculty Fellow   Center for Science, Law & Innovation
Glenn.Hoetker@asu.edu<mailto:Glenn.Hoetker@asu.edu>  | http://hoetker.faculty.asu.ed<http://hoetker.faculty.asu.edu>u | 480-965-4566


[cid:A94BE2E2-5D9A-439B-9D2E-34434B2F9153]

On Jun 21, 2017, at 8:31 PM, amaxwell <notifications@github.com<mailto:notifications@github.com>> wrote:


It should time out, eventually. The server validation is probably trying a mirror that's unavailable. How long did you wait for "validating server?"

**Comment by ghoetker** [comment]
Hmm.  Changed the repository I had set for home, quit, restarted and then it worked. Tried the same yesterday without success.

Odd.

However, it seems like problem solved, at least as far as within TLU’s control.

Thank you very much.

G.
Glenn P. Hoetker
Arizona State University | W. P. Carey School of Business
Professor & Dean's Council Distinguished Scholar
Affiliate Professor   Sandra Day O'Conner College of Law
Senior Sustainability Scholar   Julie Ann Wrigley Global Institute of Sustainability
Faculty Fellow   Center for Science, Law & Innovation
Glenn.Hoetker@asu.edu<mailto:Glenn.Hoetker@asu.edu>  | http://hoetker.faculty.asu.ed<http://hoetker.faculty.asu.edu>u | 480-965-4566


[cid:A94BE2E2-5D9A-439B-9D2E-34434B2F9153]

On Jun 22, 2017, at 3:04 PM, Glenn Hoetker <Glenn.Hoetker@asu.edu<mailto:Glenn.Hoetker@asu.edu>> wrote:

Hi. Tried again today. After 2 hours, it was still stuck.

Glenn

Glenn P. Hoetker
Arizona State University | W. P. Carey School of Business
Professor & Dean's Council Distinguished Scholar
Affiliate Professor   Sandra Day O'Conner College of Law
Senior Sustainability Scholar   Julie Ann Wrigley Global Institute of Sustainability
Faculty Fellow   Center for Science, Law & Innovation
Glenn.Hoetker@asu.edu<mailto:Glenn.Hoetker@asu.edu>  | http://hoetker.faculty.asu.ed<http://hoetker.faculty.asu.edu/>u | 480-965-4566


<email_signature_111816.png>

On Jun 21, 2017, at 8:31 PM, amaxwell <notifications@github.com<mailto:notifications@github.com>> wrote:


It should time out, eventually. The server validation is probably trying a mirror that's unavailable. How long did you wait for "validating server?"

**Comment by amaxwell** [comment]
Glenn, if this happens again, please take a sample using Activity Monitor or in Terminal while it's hanging. There's no way it should hang for that long, obviously!

**Comment by ghoetker** [comment]
Will do!  Thanks.

Sent from my iPhone

On Jun 23, 2017, at 12:12 PM, amaxwell <notifications@github.com<mailto:notifications@github.com>> wrote:

Glenn, if this happens again, please take a sample using Activity Monitor or in Terminal while it's hanging. There's no way it should hang for that long, obviously!

**Comment by ganong123** [comment]
I just encountered the exact same problem.

@ghoetker you wrote "Changed the repository I had set for home"

Can you provide some detail what you changed the repo to? (I've already done all the Schultz instructions for updating to El Capitan/Sierra.)

Thanks

**Comment by amaxwell** [comment]
@ganong123 Please take a sample using Activity Monitor while it is stuck on "Validating Server." 

Next, see if you can access TLU's Log window via its Windows menu, and post all of the output from that (select all, copy, paste).

**Comment by ghoetker** [comment]
@ganong123 unfortunately I'm on travel and away from my computer for several weeks, so I can't check for sure.  I selected one of the other US repositories and it worked.  Not sure if there is a causal relationship.

**Comment by kcmckell** [comment]
On OsX 10.12 Sierra.  I think it may be a permission mismatch.

The permissions for `texlive.tlpdb` are `-rw-------`

Here is my TLU Log:
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment initialize][695]	Welcome to TeX Live Utility 1.28, running under Mac OS X Version 10.12.5 (Build 16F73) with 4/4 processors active and 16.0 GB physical memory.
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment _checkProcessUmask][695]	Process umask = 022
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment updateEnvironment][695]	Setting up a new environment for /usr/local/texlive/2017…
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment updatePathEnvironment][695]	Using PATH = "(
    "/usr/bin",
    "/bin",
    "/usr/sbin",
    "/sbin",
    "/Library/TeX/texbin"
)"
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment _checkSystemPythonMajorVersion:minorVersion:][695]	Using python at '/usr/bin/python'
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment _checkSystemPythonMajorVersion:minorVersion:][695]	Python version is 2.7.10 (default, Feb  7 2017, 00:08:15) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment updateEnvironment][695]	Nothing to update for /usr/local/texlive/2017
2017-07-06 21:13:19 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	Checking the repository version.  Please be patient.
2017-07-06 21:13:19 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	Downloading at least 2048 bytes of tlpdb for a version check…
2017-07-06 21:13:19 +0000 Notice -[TLMDatabase connection:didFailWithError:][695]	Failed to download tlpdb for version check file:///usr/local/texlive/2017/ : Error Domain=NSURLErrorDomain Code=-1102 "You do not have permission to access the requested resource." UserInfo={NSUnderlyingError=0x60800024f4b0 {Error Domain=kCFErrorDomainCFNetwork Code=-1102 "You do not have permission to access the requested resource." UserInfo={NSErrorFailingURLStringKey=file:///usr/local/texlive/2017/tlpkg/texlive.tlpdb, NSLocalizedDescription=You do not have permission to access the requested resource., NSErrorFailingURLKey=file:///usr/local/texlive/2017/tlpkg/texlive.tlpdb}}, NSErrorFailingURLStringKey=file:///usr/local/texlive/2017/tlpkg/texlive.tlpdb, NSErrorFailingURLKey=file:///usr/local/texlive/2017/tlpkg/texlive.tlpdb, NSLocalizedDescription=You do not have permission to access the requested resource.}
2017-07-06 21:13:19 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	Downloaded 0 bytes of tlpdb for version check
2017-07-06 21:13:19 +0000 Notice -[TLMDatabase packageNamed:][695]	*** WARNING *** Requested package 00texlive.config from database without a full download
2017-07-06 21:13:19 +0000 Notice -[TLMDatabase texliveYear][695]	Unable to determine year from 00texlive.config
2017-07-06 21:13:19 +0000 Notice -[TLMEnvironment initWithInstallDirectory:][695]	Failed to determine local TeX Live version information.  This is a very bad sign.
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment _logEnvironment][695]	Current environment from /usr/bin/env:
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.sIwOa0hOMC/Render
CC_FOR_BUILD=no_compiler_found
HOME=/Users/kmckell
LOGNAME=kmckell
PATH=/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin
SHELL=/bin/bash
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.tOycCByOg3/Listeners
TMPDIR=/var/folders/9m/gzmrds4115xghmh9ls0k3q7m0000gn/T/
USER=kmckell
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
2017-07-06 21:13:19 +0000 Notice -[TLMEnvironment _checkForRootPrivileges][695]	Root permission required for installation at /usr/local/texlive/2017
2017-07-06 21:13:19 +0000 Notice +[TLMEnvironment updateEnvironment][695]	Nothing to update for /usr/local/texlive/2017
2017-07-06 21:13:19 +0000 Notice -[TLMInstallDataSource _loadRootNode][695]	Using previous profile for TeX Live 2016 from /Users/kmckell/Library/Application Support/TeX Live Utility/default.tluprofile
2017-07-06 21:13:19 +0000 Notice -[TLMMainWindowController showWindow:][695]	User has chosen to permanently ignore the GPG install alert
2017-07-06 21:14:33 +0000 Notice -[TLMLogWindowController awakeFromNib][695]	Loaded log window controller
2017-07-06 21:17:37 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	Checking the repository version.  Please be patient.
2017-07-06 21:17:37 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	Downloading at least 2048 bytes of tlpdb for a version check…
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase connection:didFailWithError:][695]	Failed to download tlpdb for version check ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet : Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x60800025e300 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "The request timed out." UserInfo={NSErrorFailingURLStringKey=ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet/tlpkg/texlive.tlpdb, NSErrorFailingURLKey=ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet/tlpkg/texlive.tlpdb, _kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4, NSLocalizedDescription=The request timed out.}}, NSErrorFailingURLStringKey=ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet/tlpkg/texlive.tlpdb, NSErrorFailingURLKey=ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet/tlpkg/texlive.tlpdb, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet/tlpkg/texlive.tlpdb took more than 30 seconds to respond.  Cancelling request.
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase _downloadDatabaseHead][695]	Downloaded 0 bytes of tlpdb for version check
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase packageNamed:][695]	*** WARNING *** Requested package 00texlive.config from database without a full download
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase texliveYear][695]	Unable to determine year from 00texlive.config
2017-07-06 21:18:08 +0000 Notice -[TLMMainWindowController _refreshUpdatedPackageListFromLocation:][695]	Not updating package list, since the repository database version could not be determined
2017-07-06 21:18:08 +0000 Notice -[TLMMirrorController _handleVersionCheckNotification:][695]	{
    URL = "ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet";
    year = "-1";
}
2017-07-06 21:18:08 +0000 Notice -[TLMMirrorController _handleVersionCheckNotification:][695]	repository = {
	type = 3;
	value = "ftp://ftp.math.purdue.edu/mirrors/ctan.org/systems/texlive/tlnet";
}
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase texliveYear][695]	Failed to use this repository 0.1 seconds ago.  Using that result.
2017-07-06 21:18:08 +0000 Notice -[TLMMainWindowController _refreshUpdatedPackageListFromLocation:][695]	Not updating package list, since the repository database version could not be determined
2017-07-06 21:18:08 +0000 Notice -[TLMDatabase texliveYear][695]	Failed to use this repository 0.1 seconds ago.  Using that result.
2017-07-06 21:18:08 +0000 Notice -[TLMMainWindowController _refreshUpdatedPackageListFromLocation:][695]	Not updating package list, since the repository database version could not be determined

**Comment by amaxwell** [comment]
@kcmckell That is a permission issue that several users have seen, but I have no idea how it gets screwed up (my code never modifies the tlpdb). You'll need to fix that manually in Terminal, unfortunately.
`sudo chmod go+r /usr/local/texlive/2017/tlpkg/texlive.tlpdb`

**Comment by kcmckell** [comment]
@amaxwell Thanks for the follow-up.  If the OP is still having this issue, your `chmod` fixed it for me.
Cheers and thanks for maintaining TLU.

**Comment by amaxwell** [comment]
I've just released a new version with a workaround for the permission problem. @ganong123, try this and let me know if it fixes your problem.

**Comment by ganong123** [comment]
Updated to the new version. Bug is resolved. Thanks @amaxwell!

**Comment by amaxwell** [comment]
Did anyone who has seen this bug update using `tlmgr` via the command line? If so, what is the output of `sudo umask` in Terminal? I'm trying to figure out what caused this.

**Comment by JnsPttn** [comment]
I got the same problem and it caused an Internal Issue.

This is the issue report:

Exception Name: NSInternalInconsistencyException
Description: Invalid parameter not satisfying: _installedYear != TLMDatabaseUnknownYear
User Info: (null)

0   CoreFoundation                      0x00007fffb4c302cb __exceptionPreprocess + 171
1   libobjc.A.dylib                     0x00007fffc9a4848d objc_exception_throw + 48
2   CoreFoundation                      0x00007fffb4c35042 +[NSException raise:format:arguments:] + 98
3   Foundation                          0x00007fffb667dbe0 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 195
4   TeX Live Utility                    0x0000000105eb1639 -[TLMEnvironment _getValidServerURL:repositoryYear:fromURL:] + 181
5   TeX Live Utility                    0x0000000105eb1ae0 -[TLMEnvironment validServerURLFromURL:] + 74
6   TeX Live Utility                    0x0000000105e75570 -[TLMMainWindowController _goHome] + 275
7   TeX Live Utility                    0x0000000105e7c7d4 -[TLMMainWindowController goHome:] + 30
8   libsystem_trace.dylib               0x00007fffca5603a7 _os_activity_initiate_impl + 53
9   AppKit                              0x00007fffb2e1d721 -[NSApplication(NSResponder) sendAction:to:from:] + 456
10  AppKit                              0x00007fffb2901cc4 -[NSControl sendAction:to:] + 86
11  AppKit                              0x00007fffb2901bec __26-[NSCell _sendActionFrom:]_block_invoke + 136
12  libsystem_trace.dylib               0x00007fffca5603a7 _os_activity_initiate_impl + 53
13  AppKit                              0x00007fffb2901b44 -[NSCell _sendActionFrom:] + 128
14  AppKit                              0x00007fffb2944539 -[NSButtonCell _sendActionFrom:] + 98
15  libsystem_trace.dylib               0x00007fffca5603a7 _os_activity_initiate_impl + 53
16  AppKit                              0x00007fffb2900426 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 2481
17  AppKit                              0x00007fffb2944272 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 798
18  AppKit                              0x00007fffb28feddb -[NSControl mouseDown:] + 832
19  AppKit                              0x00007fffb2f9924f -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 6341
20  AppKit                              0x00007fffb2f95a6c -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 1942
21  AppKit                              0x00007fffb2f94f0a -[NSWindow(NSEventRouting) sendEvent:] + 541
22  AppKit                              0x00007fffb2e19681 -[NSApplication(NSEvent) sendEvent:] + 1145
23  AppKit                              0x00007fffb2694427 -[NSApplication run] + 1002
24  AppKit                              0x00007fffb265ee0e NSApplicationMain + 1237
25  TeX Live Utility                    0x0000000105e73683 main + 491
26  TeX Live Utility                    0x0000000105e73490 start + 52

I tried to get some more data but I am totally new to this so I do not know if it helps in any way:

TeX Live Utility[460]: BUG in libdispatch client: kevent[EVFILT_MACHPORT] monitored resource vanished before the source cancel handler was invoked

Error:
NSURLConnection finished with error - code -1102

I also did any analysis of TeXLiveUtility with Activity Monitor. I can upload it if it is interesting for you in some way.
